packages/graphql/lib/interfaces/schema-file-config.interface.ts

```diff
@@ -0,0 +1,40 @@
+export interface AliasDirectiveImport {
+  name: string;
+  as: string;
+}
+
+export interface Federation2Config {
+  /**
+   * The imported directives
+   * @default ['@key', '@shareable', '@external', '@override', '@requires']
+   */
+  directives?: (string | AliasDirectiveImport)[];
+  /**
+   * The import link
+   * @default 'https://specs.apollo.dev/federation/v2.0'
+   */
+  importUrl?: string;
+}
+
+export type UseFed2Value = boolean | Federation2Config;
+
+export interface SchemaFileConfig {
+  /**
+   * If enabled, it will use federation 2 schema
+   * 
+   * **Note:** You need to have installed @apollo/subgraph@^2.0.0 and enable `autoSchemaFile`
+   * 
+   * This will add to your schema:
+   * ```graphql
+      extend schema @link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@key", "@shareable", "@external", "@override", "@requires"])
+   * ```
+  */
+  useFed2?: UseFed2Value;
```

Comment:
Not to be pedantic- but wouldn't `"federationVersion"?: FederationVersion` (an enum or similar) be more flexible to future changes?

Reply:
@Teagan42 I think that we wouldn't have such a big change as we had with apollo federation 2 vs 1 in any near future. Due to to how is setup, I think would be best to just be like that

Reply:
I agree with @Teagan42 - we don't know what the future brings, but we know for certain that Fed V2 won't be the latest version forever and we'll eventually need to deprecate a `useFed2` property. At least `federationVersion` can have the default updated to adopt the current version, and overwritten for backward compatibility (where people need it).

Reply:
If this were GraphQL specification change, no way useFed2 would hold water. In that spirit, please accommodate future development with a scalable, flexible, extendible, and maintainable pattern

Reply:
@greguintow Let me know if you're interested in making these changes, otherwise, I'll take over! 🙌

Reply:
@kamilmysliwiec I'd have time for it this weekend, if that's okay